pkg/test/framework/components/echo/echotest/run.go

```diff
@@ -61,6 +64,28 @@ func (t *T) Run(testFn oneToOneTest) {
 	})
 }
 
+// RunFromClusters will generate and run one subtest to send traffic to
+// destination instance. This is for ingress gateway testing when source instance
+// is sending requests to destination through the ingress gateway.
```

Comment:
Can we make the ingress gateway more of an example: 

```suggestion
// destination instances. This can be used when we're not using echo workloads
// as the source of traffic, such as from the ingress gateway. For example:
//    
//    RunFromClusters(func(t framework.TestContext, src cluster.Cluster, dst echo.Instances)) {
//      ingr := ist.IngressFor(src)
//      ingr.CallWithRetryOrFail(...)
//    })
```